documentation/release_notes.markdown

```diff
@@ -61,179 +94,253 @@ This release retires all api versions before v4 and promotes v4 to be
+  with that node i.e. `/pdb/query/v4/ndoes/foo.com/facts`
+  ([PDB-1473](https://tickets.puppetlabs.com/browse/PDB-1473))
 
-* Retire `[database]` configuration key node-ttl-days ([PDB-644](https://tickets.puppetlabs.com/browse/PDB-644))
+* To improve performance on the `aggregate-event-counts` endpoint, we've modified the API and underlying storage indexing. We've also dropped HSQLDB support for this endpoint. ([PDB-1587](https://tickets.puppetlabs.com/browse/PDB-1587))
 
-* Retire `[jetty]` configuration key certificate-whitelist ([PDB-886](https://tickets.puppetlabs.com/browse/PDB-886))
+#### Deprecations
 
-    This now lives in the `[puppetdb]` section.
+* HyperSQL is now deprecated and will be removed in the next major release ([PDB-1508](https://tickets.puppetlabs.com/browse/PDB-1508))
 
-* PuppetDB now uses the Ezbake build system ([PDB-663](https://tickets.puppetlabs.com/browse/PDB-663))
-
-    * The old system is still around, until we work out how to do source
-      based installations of PuppetDB in this new world. This will be done as
-      another piece of work separately.
-    * Use the nrepl trapperkeeper service. We provide a warning if
-      users use the old `[repl]` block. Instead users must define the
-      `[nrepl]` block, and it is no longer possible to choose a `type`
-      of repl service.
-    * Binaries are now installed in /usr/bin instead of /usr/sbin as per the ezbake
-      standards.
-    * Lots of template files were transferred into resource/ext, this is what ezbake
-      expects.
-    * Documentation has moved inside /usr/share/puppetdb/docs to a new location
-      depending on the distribution. This is to follow the ezbake standards.
-    * Many symlinks were removed from /usr/share/puppetdb that were pointing
-      at the /var/lib/puppetdb and /etc/puppetdb/ spaces. This finishes a partially
-      completed migration from the past.
+    Installing via the package will now no longer default to using HyperSQL.
+    Instead now we expect PostgreSQL by default, and will return an error if
+    it hasn't been configured.
 
 ### New Features
 
-* Add support for hyphenated classnames, supported by Puppet 2.7.x - 3.7.x. ([PDB-1024](https://tickets.puppetlabs.com/browse/PDB-1024))
+* We now store report metrics ([PDB-1192](https://tickets.puppetlabs.com/browse/PDB-1192))
 
-* Add a '-v' flag to the CLI to get the PuppetDB version.
+* We now store report logs ([PDB-1192](https://tickets.puppetlabs.com/browse/PDB-1184))
 
-* The `reports` database table was using the hash string as its
-  primary key, we know have switched to using a smaller bigint primary
-  key for that table, which should result in faster joins in most
-  cases, and hopefully smaller foreign key indexes sizes for tables
-  relating to
-  `reports`. ([PDB-1218](https://tickets.puppetlabs.com/browse/PDB-1218))
+* We have added `count` aggregation capabilities to the query API ([PDB-1181](https://tickets.puppetlabs.com/browse/PDB-1181))
```

Comment:
Talking about function as a feature seems a little odd here as count is the only function we support. I'll add group_by